builtin/mainmenu/tab_multiplayer.lua

```diff
@@ -231,6 +262,102 @@ local function main_button_handler(tabview, fields, name, tabdata)
 		core.start()
 		return true
 	end
+	if fields.btn_mp_search then
+		tabdata.fav_selected = 1
+		local input = fields.te_search:lower()
+		tabdata.search_for = fields.te_search
+		if input == "" then 
+			menudata.search_result = nil
+			return true 
+		end
+		if #menudata.favorites < 2 then return true end
+		
+		menudata.search_result = {}
+		
+		-- setup the filters
+		local filters = {}
+		local search_for = ""
+		for word in input:gmatch("%S+") do
+			if word:find(":") then
+				local filter = word:sub(1,string.find(word,":")-1)
+				local setting = word:sub(string.find(word,":")+1,-1)
```

Comment:
With this you call `string.find(word, ":")` three times. Using a variable would tidy up this a bit IMO.